accounts/models.py

```diff
@@ -220,7 +220,30 @@ def get_email_appellation(self):
 
         return _("user")
 
+    def clean(self):
+        super(User, self).clean()
+        qset = self.__class__.objects.filter(email__iexact=self.email)
+        if self.pk is not None:
+            # In case editing an existing user object
+            qset = qset.exclude(pk=self.pk)
+        if qset.exists():
+            from django.core.exceptions import ValidationError
+            raise ValidationError(
+                {"email": _("That email address is already in use.")})
+
     def save(self, *args, **kwargs):
+        update_fields = kwargs.get("update_fields")
+
+        # This is for backward compatibility.
+        # Because user instances are frequently updated when auth_login,
+        # reset_password. Without this, all user won't be able to login.
```

Comment:
Phrasing: "all user won't be able to login." -> one of:

* no user will be able to log in
* some users will be unable to log in.